### ISSUE TITLE
Add declaration-block-no-noncompliant-contrast rule

### ISSUE THREAD

**Issue by davidtheclark** [stylelint/stylelint#786]
This rule could use [postcss-contrastguard](https://github.com/jonathantneal/postcss-contrastguard) to ensure WCAG compliant contrasts between background and text colors.

It would have the same limitations as contrastguard, which are kind of inescapable, unfortunately (e.g. can't know what the `background` is unless it's declared in the same rule as the `color`).
##

**Comment by davidtheclark** [comment]
Idea from @ai in https://github.com/stylelint/stylelint/issues/781.

**Comment by jonathantneal** [comment]
I haven’t published my plugin. Should I rename it, @davidtheclark? `postcss-wcag-contrast` seems like a more accurate name. Thoughts?

**Comment by davidtheclark** [comment]
I do like that name more.

**Comment by jeddy3** [comment]
> `postcss-wcag-contrast`

Sounds great to me.

For the `stylelint` name, how's about `color-no-wcag-noncompliant-contrast` or `color-no-inaccessible-contrast`? 

It's attached to the `color` type and what we're doing here is disallowing the use of wcag noncompliant contrast. Like the other new rules, I think this should be a `no` rule as it doesn’t make sense to enforce the opposite i.e. requiring that all colours be non compliant.

**Comment by davidtheclark** [comment]
Name accepted. Thanks.

**Comment by ai** [comment]
Can I help with this issue somehow?

**Comment by davidtheclark** [comment]
This is open to a pull request. It can be modeled on other rules that use external plugins, like `no-browser-hacks` and `no-indistinguishable-colors`.

**Comment by jeddy3** [comment]
Also, you can use this [recent PR](https://github.com/stylelint/stylelint/pull/1048) as a blue print for exactly what files are touched when a new rule is added. All of which is documented in the [developer guide](https://github.com/stylelint/stylelint/blob/master/docs/developer-guide/rules.md#creating-new-rules).

**Comment by davidtheclark** [comment]
Closing this due to apparent lack of interest (nobody built it or commented). This rule is an excellent plugin candidate.
